service/worker/esanalyzer/workflow.go

```diff
@@ -0,0 +1,410 @@
+	endTime := time.Now().Add(
+		-time.Nanosecond * time.Duration((int64(info.Duration.AvgExecTimeNanoseconds) * 3)),
+	).UnixNano()
+	if endTime > maxEndTimeAllowed {
+		endTime = maxEndTimeAllowed
+	}
+
+	maxNumWorkflows := analyzer.config.ESAnalyzerNumWorkflowsToRefresh(info.Name)
+	query, err := getFindStuckWorkflowsQuery(startDateTime, endTime, info.Name, maxNumWorkflows)
+	if err != nil {
+		logger.Error("Failed to create ElasticSearch query for stuck workflows",
+			zap.String("error", fmt.Sprintf("%v", err)),
+			zap.String("startDateTime", fmt.Sprintf("%v", startDateTime)),
+			zap.String("endTime", fmt.Sprintf("%v", endTime)),
+			zap.String("workflowType", info.Name),
+			zap.String("maxNumWorkflows", fmt.Sprintf("%v", maxNumWorkflows)),
+		)
+		return nil, err
+	}
+	response, err := analyzer.esClient.SearchRaw(ctx, analyzer.visibilityIndexName, query)
+	if err != nil {
+		logger.Error("Failed to query ElasticSearch for stuck workflows",
+			zap.String("error", fmt.Sprintf("%v", err)),
+			zap.String("VisibilityQuery", query),
+		)
+		return nil, err
+	}
+
+	// Return a simpler structure to reduce activity output size
+	workflows := []WorkflowInfo{}
+	if response.Hits.Hits != nil {
+		for _, hit := range response.Hits.Hits {
+			workflows = append(workflows, WorkflowInfo{
+				DomainID:   hit.DomainID,
+				WorkflowID: hit.WorkflowID,
+				RunID:      hit.RunID,
+			})
+		}
+	}
+
+	if len(workflows) > 0 {
+		analyzer.metricsClient.AddCounter(
+			metrics.ESAnalyzerScope,
+			metrics.ESAnalyzerNumStuckWorkflowsDiscovered,
+			int64(len(workflows)))
+	}
+
+	return workflows, nil
+}
+
+func getWorkflowTypesQuery(startDateTime int64, aggregationKey string) string {
+	return fmt.Sprintf(`
+    {
+      "query": {
+         "bool": {
+            "must": [
+              {
+                "range" : {
+                  "StartTime" : {
+                    "gte" : "%d"
+                  }
+                }
+              },
+              {
+                "exists": {
+                  "field": "CloseTime"
+                }
+              }
+            ]
+         }
+      },
+      "size": 0,
+      "aggs" : {
+          "%s" : {
+              "terms" : { "field" : "WorkflowType", "size": 10 },
+              "aggs": {
+                  "duration" : {
+                    "avg" : {
+                      "script" : "(doc['CloseTime'].value - doc['StartTime'].value)"
+                    }
+                  }
+              }
+          }
+      }
+    }
+	`, startDateTime, aggregationKey)
+}
+
+// getWorkflowTypes is activity to get workflow type list from ElasticSearch
+func getWorkflowTypes(ctx context.Context) ([]WorkflowTypeInfo, error) {
+	logger := activity.GetLogger(ctx)
+	analyzer := ctx.Value(analyzerContextKey).(*Analyzer)
+	limitToTypes := analyzer.config.ESAnalyzerLimitToTypes()
+	if len(limitToTypes) > 0 {
+		results := []WorkflowTypeInfo{}
+		workflowTypes := strings.Split(limitToTypes, ",")
```

Comment:
hmm.  they could contain commas, though it's probably not all that likely.
csv parser maybe?  or we could require `["a", "json", "array"]`.  though this is probably fine for now.